Dockerfile

```
@@ -4,23 +4,31 @@ FROM python:3.10-slim
     && rm -rf /var/lib/apt/lists/*
 
-# Copy requirements and install Python dependencies
+# Upgrade pip, setuptools, and wheel FIRST
+RUN python -m pip install --no-cache-dir --upgrade pip setuptools wheel
+
+# Install Python dependencies with PEP 517 disabled (for old setup.py packages)
 COPY requirements.txt .
-RUN python -m pip install --upgrade pip && \
-    python -m pip install -r requirements.txt
+RUN python -m pip install --no-cache-dir --no-use-pep517 -r requirements.txt
+RUN python -m pip install --no-cache-dir mysqlclient
```

Hadolint (2.12.0)

[info] Multiple consecutive `RUN` instructions. Consider consolidation.

(DL3059)